### ISSUE TITLE
More lerchphi strangeness

### ISSUE THREAD

**Issue by ness01** [mpmath/mpmath#252]
```
I am not quite a sure about this bug report as about the previous one. According to my computations, when passing across the branch cut from 1 to oo from above, on the standard branch, the lerch transcendent should pick up a term 2*pi*I*log(z)**(s-1)/gamma(s)/z**a [where the standard choice of argument is made for z and also for log(z)], at least for re(a) > 0. Indeed:

In [1]: def a(z, s, a, eps=1e-15): return mpmath.lerchphi(z + eps*1j, s, a) - mpmath.lerchphi(z - eps*1j, s, a)
In [2]: def b(z, s, a): return (2*pi*I*log(z)**(s-1)/z**a/gamma(s)).n()

In [3]: a(1.7, 5.5, 3.5) - b(1.7, 5.5, 3.5)
Out[3]: 2.16840434497101e-19⋅ⅈ
In [8]: a(2.5, 1.5, 3.9) - b(2.5, 1.5, 3.9)
Out[8]: 2.77555756156289e-17⋅ⅈ
In [9]: a(2.5, 1.5+1j, 3.9) - b(2.5, 1.5+1j, 3.9)
Out[9]: -1.38777878078145e-17 + 5.55111512312578e-17⋅ⅈ

However:

In [10]: a(2.5, 1.5, 3.9+1j) - b(2.5, 1.5+1j, 3.9+1j)
Out[10]: -0.55002171991196 + 0.166746405436317⋅ⅈ

Indeed something weird is going on:

In [42]: a(2.5, 1.5, 3.9 - 1e-10j, 1e-15)
Out[42]: (0.4646322489177 + 0.19040815320768j)
In [43]: a(2.5, 1.5, 3.9 + 1e-10j, 1e-15)
Out[43]: (-0.4646322489177 + 0.19040815320768j)
In [44]: a(2.5, 1.5, 3.9 + 0*1e-10j, 1e-15)
Out[44]: (0.0 + 0.19040815319107j)

[i.e. lerchphi does not seem to be continuous in a]

Increasing mpmath.mp.prec to 500 does not make this go away.

Now I must admit that I am not entirely sure how the lerch transcendent should or could be continued analytically, and even less sure about what is going on when computing it numerically. However, I think that

1) lerchphi(z, s, a) should be continuous in a, certainly for fixed s and z and re(a) > 0.
2) the branching behaviour I mentioned first should hold

In summary, it seems that for non-real a, lerchphi is evaluated on a non-principal branch.
```

Original issue for #252: http://code.google.com/p/mpmath/issues/detail?id=212
Original author: https://code.google.com/u/104531927090589914088/

**Comment by s-henao-castellanos** [comment]
Hello, I have encountered issues with the lerchphi function as well, even with high dps, at |z|>>1. 

I just wanted to leave a comment here, in the high |z| regime the hypergeometric representation of this function works well with the 3F2 function of this same package:

```python
def lerchphi_hyp(z,s,a):
    return hyp3f2(1,a,a,1+a,1+a,z) / a**s
```

I am working with |z|~60 000, s~2, and |a|~3, and the coincidence with Mathematica is good.

Maybe we can test which values or regimes work and implement branches for the adequate algorithm to be used.

**Comment by skirpichev** [comment]
It looks using Laplace transform integral like FLINT does (see the [implementation](https://fredrikj.net/blog/2022/02/computing-the-lerch-transcendent/#section6) from @fredrik-johansson blog) is better in this regime.  BTW Crandall algorithm also works:
```
In [31]: def a2(z, s, a, eps=1e-15): return lerch_hankel(z + eps*1j, s, a) - lerch_hankel(z - eps*1j, s, a)

In [32]: def a3(z, s, a, eps=1e-15): return lerch_crandall(z + eps*1j, s, a) - lerch_crandall(z - eps*1j, s, a)

In [33]: a2(2.5, 1.5, 3.9 - 1e-10j, 1e-15)
Out[33]: (-1.744693278737941e-11 + 0.19040815319107374j)

In [34]: a2(2.5, 1.5, 3.9 + 1e-10j, 1e-15)
Out[34]: (1.744693278737941e-11 + 0.19040815319107374j)

In [35]: a2(2.5, 1.5, 3.9 + 0*1e-10j, 1e-15)
Out[35]: (0.0 + 0.19040815319107376j)

In [36]: a3(2.5, 1.5, 3.9 - 1e-10j, 1e-15)
Out[36]: (-1.7446918909591602e-11 + 0.19040815319107357j)

In [37]: a3(2.5, 1.5, 3.9 + 1e-10j, 1e-15)
Out[37]: (1.7446918909591602e-11 + 0.19040815319107357j)

In [38]: %time a2(2.5, 1.5, 3.9 - 1e-10j, 1e-15)
CPU times: user 423 ms, sys: 84 μs, total: 423 ms
Wall time: 424 ms
Out[38]: (-1.744693278737941e-11 + 0.19040815319107374j)

In [39]: %time a3(2.5, 1.5, 3.9 - 1e-10j, 1e-15)
CPU times: user 581 ms, sys: 97 μs, total: 581 ms
Wall time: 581 ms
Out[39]: (-1.7446918909591602e-11 + 0.19040815319107357j)

In [40]: %time a(2.5, 1.5, 3.9 - 1e-10j, 1e-15)
CPU times: user 409 ms, sys: 233 μs, total: 409 ms
Wall time: 521 ms
Out[40]: (0.46463224891769683 + 0.19040815320767987j)
```
That match also Mathematica output.

I think we could replace current integral path with something like lerch_hankel().
